src/components/base/outline-google-map/outline-google-map.ts

```diff
@@ -0,0 +1,248 @@
+import { LitElement, html, CSSResultGroup } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import { GoogleMapsApi } from './src/google-maps-api';
+import './src/google-maps-api';
+import { OutlineMapMarker } from './src/outline-map-marker';
+import { MapSelector } from './src/map-selector';
+import './src/map-selector';
+import componentStyles from './outline-google-map.css.lit';
+
+export type MapType = 'roadmap' | 'hybrid' | 'satellite' | 'terrain';
+export type GestureType = 'auto' | 'cooperative' | 'greedy' | 'none';
```

Comment:
Minor suggestion:

If you use the pattern below you can import the list of options in the story so you only need to define it once.
```javascript
export const buttonSizes = ['small', 'medium', 'large'] as const;
export type ButtonSize = typeof buttonSizes[number];
```

Reply:
I took a stab at this and I'm not sure it's properly enforcing the type. I use:
`export const gestureTypes = ['auto', 'cooperative', 'greedy', 'none'];
export type GestureType = typeof gestureTypes[number];`

I can then set without errors:
`  @property({ type: String })
  gestureHandling: GestureType = 'green';`

Reply:
I think you need this, with `const` at the end.

```javascript
export gestureTypes = ['auto', 'cooperative', 'greedy', 'none'] as const;
```

I don't remember the specifics, but I think what you have tells it that you have an array of strings, but this says the array is a constant and so it has to be one of those values.

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions